src/Tokens/Models/UserIdToken/UserIdToken.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Pinnacle\OpenIdConnect\Tokens\Models\UserIdToken;
+
+use GuzzleHttp\Psr7\Uri;
+use Pinnacle\OpenIdConnect\Tokens\Exceptions\InvalidUserIdTokenException;
+use Pinnacle\OpenIdConnect\Tokens\Exceptions\MissingRequiredClaimKeyException;
+use Pinnacle\OpenIdConnect\Tokens\Exceptions\UserIdTokenHasExpiredException;
+use Pinnacle\OpenIdConnect\Tokens\Models\UserIdToken\Constants\ClaimKey;
+
+class UserIdToken
+{
+    /**
+     * @var array
+     */
+    private array             $parsedValues;
+
+    private Uri               $issuerIdentifier;
+
+    private SubjectIdentifier $subjectIdentifier;
+
+    private Audiences         $audiences;
+
+    private int               $expirationTime;
+
+    private int               $issuedTime;
```

Comment:
I think these should be `DateTimeInterface` types instead (with a concrete type of `DateTimeImmutable`).